CMakeLists.txt

```diff
@@ -508,6 +508,23 @@ endif()
 #### Binary ######################
 ##################################
 
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}")
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY_DEBUG ${CMAKE_RUNTIME_OUTPUT_DIRECTORY})
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY_RELEASE ${CMAKE_RUNTIME_OUTPUT_DIRECTORY})
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY_MINSIZEREL ${CMAKE_RUNTIME_OUTPUT_DIRECTORY})
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY_RELWITHDEBINFO ${CMAKE_RUNTIME_OUTPUT_DIRECTORY})
+
+if(WIN32 OR APPLE)
+    set(OD_BUILD_SUFFIX "_d" CACHE STRING "String holding a suffix appended to the name of output binaries (under CMake build, only used for debug).")
+else()
+    set(OD_BUILD_SUFFIX "" CACHE STRING "String holding a suffix appended to the name of output binaries (under CMake build, only used for debug).")
+endif()
+
+if(OD_BUILD_SUFFIX)
+    set(OD_HAS_BUILD_SUFFIX TRUE)
```

Comment:
FMI, why do you use a boolean here after testing OD_BUILD_SUFFIX ? Wouldn't testing OD_BUILD_SUFFIX later instead of the boolean be better ?